server/data/applications.ts

```diff
@@ -0,0 +1,298 @@
+     * @param {number} salary Position salary 
+     * @param {string} jobPostUrl URL the application was found at
+     * @param {string} cardColor Chosen color of application card
+     * @param {string} description Application description
+     * @returns {Promise<object>} Returns the created application object if successful anf throws
+     * and error otherwise
+     */
+    create: async (cycleId: string, company: string, position: string, location: string,
+        salary: number, jobPostUrl: string, cardColor: string, description: string):
+        Promise<object> => {
+        if(!cycleId || !checkObjectId(cycleId)) {
+            throw new Error("A proper cycle id must be provided.");
+        }
+
+        if(!company || !checkNonEmptyString(company)) {
+            throw new Error("A company name must be provided.");
+        }
+
+        if(!position || !checkNonEmptyString(position)) {
+            throw new Error("A position name must be provided.");
+        }
+
+        if(!location || !checkNonEmptyString(location)) {
+            throw new Error("A location must be provided.");
+        }
+
+        if(!salary || !checkPositiveNumber(salary)) {
+            throw new Error("A salary must be provided.");
+        }
+        if(!jobPostUrl || !checkNonEmptyString(jobPostUrl)) {
+            throw new Error("An application url must be provided.");
+        }
+
+        if(!cardColor || !checkNonEmptyString(cardColor)) {
+            throw new Error("A card color must be selected.");
+        }
+
+        if(!description || !checkNonEmptyString(description)) {
+            throw new Error("A description must be provided.");
+        }
+        
+        //Unsure of how the logo is getting stored if we are using that auto logo finder
+        let newApp: object = {
+            _id: new ObjectId(),
+            company: company,
+            position: position,
+            logo: null,
+            location: location,
+            salary: salary, 
+            jobPostUrl: jobPostUrl,
+            cardColor: cardColor,
+            description: description,
+            progress: 0,
+            notes: [],
+            events: [],
+            contacts: []
+        }
+
+        //Finds the cycle with cycleId
+        const cycleCollection: any = await cycles();
+        const cycle: object = await cycleCollection.findOne({
+            _id: new ObjectId(cycleId)
+        });
+        if(cycle === null) {
+            throw new Error("There is no cycle with this id.")
+        }
+
+        //Adds the newly created application to the cycle
+        cycle["applications"] = cycle["applications"].push(newApp);
+        let updatedInfo = await cycleCollection.updateOne({_id: new ObjectId(cycleId)}, {$set: cycle});
+        if(updatedInfo.modifiedCount === 0) {
+            throw new Error("Could not update cycle with new application.");
+        }
+
+        //Re-finds the newly updated cycle with cycleId
```

Comment:
Why do you re-find the cycle and then return the newest app from the cycle? Why not just return the object since the cycle was already checked for and you checked if the cycle collection was properly updated

Reply:
Yeah I thought it was good practice to double check but it doesn't seem necessary so I removed it